packages/kbn-test/src/functional_test_runner/lib/suite_tracker.ts

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import fs from 'fs';
+import { dirname, relative, resolve } from 'path';
+
+import { REPO_ROOT } from '@kbn/dev-utils';
+
+import { Lifecycle } from './lifecycle';
+
+export interface SuiteInProgress {
+  startTime?: Date;
+  endTime?: Date;
+  success?: boolean;
+}
+
+export interface SuiteWithMetadata {
+  config: string;
+  file: string;
+  tag: string;
+  title: string;
+  startTime: Date;
+  endTime: Date;
+  duration: number;
+  success: boolean;
+  leafSuite: boolean;
+}
+
+const getTestMetadataPath = () => {
+  return process.env.TEST_METADATA_PATH || resolve(REPO_ROOT, 'target', 'test_metadata.json');
```

Comment:
I think this should be managed by the config service so that each config could write to a separate file, it can still default to this path/env var, but that service is supposed to control stuff like this and make it configurable for consumers.

Reply:
If it was configurable by the FTR config, I think that we would at least need to have a way to force the path in CI, because we need to know where it will end up. If we're always forcing the path in CI, I'm not sure that there ends up being much benefit in making it configurable by config?

Can this be something that we change later if it ever comes up as a need?